flutter/lib/ui/painting.dart

```diff
@@ -280,20 +314,111 @@ class Paint {
   ///
   /// This color is not used when compositing. To colorize a layer, use
   /// [colorFilter].
-  Color color = _kDefaultPaintColor;
-  static const Color _kDefaultPaintColor = const Color(0xFF000000);
+  Color get color {
+    if ((_data.getInt32(_kColorOffset, Endianness.LITTLE_ENDIAN) & _kColorBit) == 0)
+      return const Color(0xFF000000);
+    return new Color(_data.getInt32(_kColorOffset, Endianness.LITTLE_ENDIAN));
+  }
+  set color(Color value) {
+    assert(value != null);
+    _data.setInt32(_kColorOffset, value.value, Endianness.LITTLE_ENDIAN);
+    _updateNullityMask(_kColorBitIndex, true);
+  }
+
+  static const _kTransferModeBit = 1 << _kTransferModeBitIndex;
+
+  /// A transfer mode to apply when a shape is drawn or a layer is composited.
+  ///
+  /// The source colors are from the shape being drawn (e.g. from
+  /// [Canvas.drawPath]) or layer being composited (the graphics that were drawn
+  /// between the [Canvas.saveLayer] and [Canvas.restore] calls), after applying
+  /// the [colorFilter], if any.
+  ///
+  /// The destination colors are from the background onto which the shape or
+  /// layer is being composited.
+  ///
+  /// If null, defaults to [TransferMode.srcOver].
+  TransferMode get transferMode {
+    if ((_data.getInt32(_kNullityMaskOffset, Endianness.LITTLE_ENDIAN) & _kTransferModeBit) == 0)
+      return TransferMode.srcOver;
+    return TransferMode.values[_data.getInt32(_kTransferModeOffset, Endianness.LITTLE_ENDIAN)];
+  }
+  set transferMode(TransferMode value) {
+    // We encode explicit TransferMode.srcOver values differently from null
+    // values because TransferMode.srcOver does not have index zero. There
+    // shouldn't be any observable effect of this encoding.
+    final int encoded = value?.index ?? 0;
```

Comment:
I see, it's because the default isn't 0.

I think I would use xor here as well. That gets rid of a bunch of logic in get, and removes the need for the bit.